js/src/Ice/ConnectionI.js

```diff
@@ -935,42 +926,24 @@ export class ConnectionI {
                 }
 
                 case StateActive: {
-                    //
-                    // Can only switch from holding or not validated to
-                    // active.
-                    //
-                    if (this._state !== StateHolding && this._state !== StateNotValidated) {
-                        return;
-                    }
+                    Debug.assert(this._state <= StateHolding);
```

Comment:
It looks like we can transition directly from StateNotValidated to StateActive, without an intermediate StateHolding.